crates/datafusion-optimizer-rules/src/logical_plan/duckdb/aggregate_pushdown.rs

```diff
@@ -0,0 +1,618 @@
+use std::sync::{Arc, LazyLock};
+
+// https://duckdb.org/docs/stable/sql/functions/aggregates
+// https://datafusion.apache.org/user-guide/sql/aggregate_functions.html
+static SUPPORTED_AGG_FUNCTIONS: LazyLock<HashSet<&str>> = LazyLock::new(|| {
+    HashSet::from([
+        // Basic aggregates
+        "avg",
+        "count",
+        "max",
+        "min",
+        "sum",
+        // Bitwise aggregates
+        "bit_and",
+        "bit_or",
+        "bit_xor",
+        // Boolean aggregates
+        "bool_and",
+        "bool_or",
+        // String aggregates
+        "string_agg",
+        // Statistical aggregates
+        "corr",
+        "covar_pop",
+        "covar_samp",
+        "median",
+        "stddev_pop",
+        "stddev_samp",
+        "var_pop",
+        "var_samp",
+        // Regression aggregates
+        "regr_avgx",
+        "regr_avgy",
+        "regr_count",
+        "regr_intercept",
+        "regr_r2",
+        "regr_slope",
+        "regr_sxx",
+        "regr_sxy",
+        "regr_syy",
+        // Percentile/quantile aggregates
+        "quantile_cont",
+        // Approximate aggregates
+        "approx_percentile_cont",
+    ])
+});
+
+/// This looks for opportunities in the expressed logical plan to push down aggregates
+/// directly into the SQL execution for `DuckDB` accelerated table providers (as indicated by `spice.accelerator`).
+///
+/// Schema metadata was chosen to "tag" scans in order to avoid a dependency on the runtime crate and
+/// concrete adapter types. This also vastly simplifies testing.
+#[derive(Debug)]
+pub struct DuckDBAggregateLogicalPushdown {}
+
+impl DuckDBAggregateLogicalPushdown {
+    #[must_use]
+    pub fn new() -> Arc<Self> {
+        Arc::new(Self {})
+    }
+
+    fn is_duckdb_provider(scan: &TableScan) -> Result<bool> {
+        let provider = source_as_provider(&scan.source)?;
+
+        Ok(matches!(
+            provider
+                .schema()
+                .metadata
+                .get("spice.accelerator")
+                .map(std::string::String::as_str),
+            Some("duckdb")
+        ))
+    }
+
+    /// If this aggregate's root scan is from a `DuckDB` accelerated source, with supported expressions,
+    /// wrap it in a marker node for pushdown rewriting during physical planning
+    #[allow(clippy::needless_continue)]
+    fn try_mark_pushdown(plan: &LogicalPlan) -> Result<Option<LogicalPlan>> {
+        // Find an aggregate node
+        let LogicalPlan::Aggregate(agg) = plan else {
+            return Ok(None);
+        };
+
+        // Validate its agg expressions to make sure they are supported
+        for expr in &agg.aggr_expr {
+            match expr {
+                Expr::AggregateFunction(AggregateFunction { func, .. })
+                    if SUPPORTED_AGG_FUNCTIONS.contains(func.name()) =>
+                {
+                    continue;
+                }
+                _ => return Ok(None),
+            }
+        }
```

Comment:
```suggestion
        if agg.aggr_expr.iter().any(|expr| {if let Expr::AggregateFunction(AggregateFunction { func, .. }) && SUPPORTED_AGG_FUNCTIONS.contains(func.name()) { false } else { true }}) {
        return Ok(None)
        }
```

would this not be better represented by an `.iter().any()`? this is checking if any item is not a supported aggregate function

Reply:
How about this? "All must be supported"?
```rust
        if !agg.aggr_expr.iter().all(|e| match e {
            Expr::AggregateFunction(AggregateFunction { func, .. }) => {
                SUPPORTED_AGG_FUNCTIONS.contains(func.name())
            }
            _ => false,
        }) {
            return Ok(None);
        }
```